docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,231 @@
+
+The server will look up the appropriate shared secret based on the token id.  If the server doesn't know about the token specified or there is no token specified it MUST respond with an HTTP 403 forbidden error.
+
+If it does recognize the token it will respond with a JWS object using the flattened JWS JSON serialization syntax (JWS section 7.2.2).
+
+The `Content-Type` header of the response MUST be `application/jose+json`. The `kid` (key id) header value MUST be the `token-id`. The `alg` MUST be `HS256` (HMAC-SHA256).
+
+The full JWS will look something like:
+
+```json
+{
+  "payload":
+    "eyJ0eXBlIjoiQ2x1c3RlckxvY2F0b3IiLCJ2ZXJzaW9uIjoiMS4wIiwiZW5kcG9p
+     bnRzIjpbIjEwLjAuMC4xIiwiMTAuMC4wLjIiLCJteWNsdXN0ZXIuZXhhbXBsZS5jb
+     20iLCIxLjIuMy40IiwiMS4yLjMuNSJdLCJyb290Q2VydGlmaWNhdGVzIjpbIk1JSU
+     RGRENDQWZ5Z0F3SUJBZ0lKQUlyL0FtbktFZGVzTUEwR0NTcUdTSWIzRFFFQkJRVUF
+     NQkF4RGpBTUJnTlZCQU1UQldwaVpXUmhNQjRYRFRFMk1EZ3hNVEl3TWpjME1Wb1hE
+     VFF6TVRJeU9ESXdNamMwTVZvd0VERU9NQXdHQTFVRUF4TUZhbUpsWkdFd2dnRWlNQ
+     TBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUURBaUNLNnQ2TGhId2
+     9qeitSUTlJVklPQTJQK21MNlhXU0ExbzZBYVNNdWR6SlJPMkdyNjBURldIb3o1aTk
+     rdnAybmdaQlJOT3dqSWMxaXlrdlY1ckZQOG1OWktCM3hYbzZYc0tYVVZEUDRtYWl5
+     T3V2eXdrdnRuaEgvSFUvNW9JbFZQREx6MkxXbk9aZDdYaXArWmdvY2Q3NTZwMjJKM
+     G9hZ21XaWVuNU9ZSHNXQWxPcWJ1dDVVaGZTSDRnRWRDL3BKK1FzODZOMjkyKzJBM1
+     R6SkMwZlg5SHJjQ1lKRVExdjI0OTFwOFJVK0czUitJRk5YUHJPdnVZMmRITi93M0J
+     LK1ZxZXR0VEt2SU5YQ2tTbXM2Tnc5aXFEanlXY1JCaE5nM2w0ajR5azV6WlVtVVEy
+     a1EwdXJYWFpNYVhTR1NPWi9ETGd5ZkVBUGJYNXJVT0wxU05lUkFnTUJBQUdqY1RCd
+     k1CMEdBMVVkRGdRV0JCUm5jb1Bwa0tQd3VCaDVvV1NFSys4b1lvMjF3ekJBQmdOVk
+     hTTUVPVEEzZ0JSbmNvUHBrS1B3dUJoNW9XU0VLKzhvWW8yMXc2RVVwQkl3RURFT01
+     Bd0dBMVVFQXhNRmFtSmxaR0dDQ1FDSy93SnB5aEhYckRBTUJnTlZIUk1FQlRBREFR
+     SC9NQTBHQ1NxR1NJYjNEUUVCQlFVQUE0SUJBUUFacEFzQmZsR3VZU2dTQ1p1akVwS
+     FJZbVArUGwvQVBlMzcraUJpb2k0SmdNU01LeWRKRnZGSDFtUWFTSlZjNnl2TmcvbX
+     FNRjlkcktiY1gvZkVkZ3grMjVEcWRYUm92eXZBVjUvUmxTUnIwUkE0eFVhbmlRczB
+     IMEQxVlVvNUg5L1ZDSDRTNUN3WVlQUEcvMXVQeUt4Z3NSV0ltM29sZE5pRTkzTnNl
+     U3pkNjVKVHdnN05NSVNnOVlNdGJwZG1ETThyclZtMThRTGp5akRkeEVvS0VmSnRha
+     GwxY0F3OFh0b3FZRGN6d01LSVR4NFZvVEUzZ3lNSThUS0l3M1lOT3pqem9rdkV6bH
+     NzY01oV3J5SWpBUmpSY0ordDlrenRQeEI2YlVFaWQ4YU9vRzNPcEdqcGpBbHByNFQ
+     zb0lXby9ZRThLMksxdGUvM2dFLzVMM1FhS00yLyJdfQ",
+  "protected":"eyJhbGciOiJIUzI1NiIsImtpZCI6IkE4MUU1ZDREd0kifQ",
+  "signature":"vat_c1aTytSLVkPw9wWhven9GtLBSLFzJViIq5scLDA"
+}
+```
+If we Base64 decode the `protected` member we get:
+```json
+{
+  "alg":"HS256",
+  "kid":"A81E5d4DwI"
+}
+```
+
+The payload is the ClusterInfo JSON object as described above.  Astute readers will note that the actual root certificates are double Base64 encoded.  It is a good thing that efficiency isn’t critical here.
+
+*Implementation Note:* there looks to be a decent library for dealing with JWS: https://github.com/square/go-jose.
+
+#### Method: ClusterInfo refresh
```

Comment:
Both can happen.  The set of API servers can migrate around and certificates can be rotated.

We could implement CRLs and OCSP but that is probably overkill and, honestly, people don't check them that often.  I'd leave that as a future problem for now and spawn a new design doc.